### ISSUE TITLE
Enable downloading music file and playing during game

### ISSUE THREAD

**Issue by circumspect** [circumspect/White-Rabbit#53]

**Comment by stephenkilpatrick** [comment]
I really like this feature. After getting white-rabbit up and running this weekend, had my first game with it, and it's the first time I ever played a role playing game that had music in it and I thought that it improved the game. Since it's text based, I think that's what makes the difference. With the Medium article, having the YouTube video with the timer showing would be nice. However, if I could add one more feature request to this - what do you think about allowing users to record their own MP3/WAV/whatever for voicemails and have them start playing at 00:00 ?

I liked having the typed out versions going into #voicemails going into the exported PDF, but I saw an actual play with them playing over the last of the soundtrack, and that was pretty nice. I'm not sure how many moving parts that adds on top of the bot just handling the YouTube video, which looks simple enough.

**Comment by circumspect** [comment]
I don't think Discord supports bots streaming video, so the only way to show the official timer currently is to have one of the players stream it. This feature would mainly be for people who couldn't stream the timer, in conjunction with the bot's ability to provide timer messages at regular intervals (see the documentation [here](https://white-rabbit.readthedocs.io/en/stable/user-guide/settings.html)).

When I was running the game, I usually had players read their voicemails out loud after the game ended. It would be pretty straightforward to add a command for the bot to collect audio voicemails and play them out loud, but I'm not sure if it's a feature that's really necessary.

**Comment by stephenkilpatrick** [comment]
One of my players was on such a poor Internet connection that having that YouTube streamed seemed to cause them issues, so having the audio only built right into the bot plus the bot just announcing the time periodically would probably have helped them out. Unfortunate that someone in 2021 not-that-rural America had that connectivity issue, but is what it is, right?

I played in a game of [10 Candles](https://cavalrygames.com/ten-candles/) some time ago and used the voicemail mechanics where we'd record the voicemails on our phones and then played the back. Seemed to carry a bit more of an emotional punch than just writing them down and reading them later, however - I think I'm with you that having the voicemails saved, and then played at the end of the game automatically probably not a top tier feature request.

On the topic of the voicemails and the PDF output, I hadn't seen in the documentation that that's how voicemails would work. I could draft up some documentation explaining how that works and push it into git a little later this week, if you'd think that'd be valuable.

**Comment by circumspect** [comment]
Ah, I'm sorry to hear that. In the meantime, you could use groovy or another music bot to play the audio if that happens in the future?

Thanks for mentioning that the documentation doesn't talk about voicemails - that's an oversight on my part. I'll add a note in the quickstart guide later today. If the PDF you generated doesn't include the voicemails and you would like them, you can have each player add their voicemail to the voicemails channel and then just call `!pdf` again.

**Comment by stephenkilpatrick** [comment]
Thanks for the recommendation.

On the topic of documentation - I wouldn't mind seeing a bit more of the whole thing fleshed out (which I'd be happy to lend my assistance to). I wasn't perfectly sure of how the whole thing runs, even with the existing documentation and the special instructions to Charlie Barnes's clues channels. How white-rabbit ran worked out how I presumed, so that's good - but I wasn't *sure* that it would.

And, I'm guessing there is a fine line between being very clear with how white-rabbit runs and how Alice is Missing runs, which probably wouldn't be good for Hunters Entertainment's bottom line.

Again, I've got some free time, I'm happy to work on some documentation, but I don't want to step on your toes, if you'd like to keep everything close. Anyway about it - you've done some fine work here.

**Comment by circumspect** [comment]
Yeah, I focused on documenting the bot as opposed to the game itself, as I assumed whoever was taking the role of Charlie would have that available and be able to explain it to the other players. And, as you said, I wanted to encourage players to buy a copy of the game to go with the bot. In general, if it was answered in the the Alice is Missing rulebook, I left it out of the documentation. The quickstart guide also has a note at the top mentioning that it assumes familiarity with the rules.

Could you clarify which parts in particular you were unsure about? I'm not sure what the documentation is missing, but I'd be happy to answer any questions you may have had and add to it.

**Comment by stephenkilpatrick** [comment]
Oh sure, I think it's just, I wasn't really sure what exactly happened after I hit !init (easy to observe after the fact, but maybe it'd be a bonus to explain that Character and Drive cards go into the clues rooms), but mainly just an explanation that the clue cards would be automatically assigned to characters at the right time *and* that if there was something that happened later or needed a random coin flip type thing, it'd just be handled by the bot. Good work on being thorough about that too.

It all works fine, just I wasn't certain if I'd have to be doing that work or preparing my players to do that for themselves, even though the bot handled it.

**Comment by circumspect** [comment]
Sure, I'll add that in when I get the chance. Thanks for your input!
